var/spack/repos/builtin/packages/neuron/package.py

```diff
@@ -0,0 +1,236 @@
+                            '--host=powerpc64',
+                            '--without-memacs'])
+
+        if 'cray' in self.spec.architecture:
+            options.extend(['--without-memacs',
+                            '--without-nmodl',
+                            'cross_compiling=yes'])
+
+            # TODO: on cray systems we get an error while linking even if
+            # we use cc wrapper. Hence add explicitly add mpi library
+            if spec.satisfies('+mpi'):
+                options.append('LIBS=%s' % spec['mpi'].libs.link_flags)
+
+            # TODO: -pthread is not a valid pthread option for cray compiler.
+            # for now disable use of pthread with cray compiler.
+            if spec.satisfies('%cce'):
+                options.append('use_pthread=no')
+
+        return options
+
+    def get_opt_flags(self):
+        if 'bgq' in self.spec.architecture:
+            return '-O3 -qtune=qp -qarch=qp -q64 -qstrict -qnohot -g'
+        else:
+            return '-O2 -g'
+
+    def get_arch_dir(self):
+        if 'bgq' in self.spec.architecture:
+            arch = 'powerpc64'
+        elif 'cray' in self.spec.architecture:
+            arch = 'x86_64'
+        else:
+            arch = self.spec.architecture.target
+
+        return arch
+
+    def get_python_options(self, spec):
+        options = []
+
+        if spec.satisfies('+python'):
+            py_version = 'python{0}'.format(spec['python'].version.up_to(2))
+            python_exec = spec['python'].command.path
+
+            options.append('--with-nrnpython=%s' % python_exec)
+            options.append('--disable-pysetup')
+
+            if spec.satisfies('+cross-compile'):
+                # on cross compile builds we have to provide
+                # PYINCDIR, PYLIB, PYLIBDIR etc
+
+                py_lib = spec['python'].prefix.lib
+                py_inc = spec['python'].headers.directories[0]
+
+                if not os.path.isdir(py_lib):
+                    py_lib = spec['python'].prefix.lib64
+
+                options.extend(['PYINCDIR=%s' % (py_inc),
+                                'PYLIB=-L%s -l%s' % (py_lib, py_version),
+                                'PYLIBDIR=%s' % py_lib,
+                                'PYLIBLINK=-L%s -l%s' % (py_lib, py_version)])
+        else:
+            options.append('--without-nrnpython')
+        return options
+
+    def get_compiler_options(self, spec):
+        options = []
+        flags = self.get_opt_flags()
+
+        # TODO: issue with static build and pgi compiler.
+        # need to add fpic and enabled-shared
+        if spec.satisfies('%pgi'):
+            flags += ' ' + self.compiler.pic_flag
+            options.extend(['CFLAGS=%s' % flags,
+                            'CXXFLAGS=%s' % flags,
+                            '--enable-shared'])
+        else:
+            options.extend(['CFLAGS=%s' % flags,
+                            'CXXFLAGS=%s' % flags])
```

Comment:
The CFLAGS/CXXFLAGS logic is repeated. If you bring it out of the if-statement, you don't need to repeat it.